test/fuzz_group.cpp

```diff
@@ -1001,6 +1010,53 @@ void parse_and_apply_instructions(std::string& in, const std::string& path, util
     }
 }
 
+struct RealmPathGuard {
+RealmPathGuard(const std::string& path) : m_path(path)
+{
+    cleanup();
+}
+~RealmPathGuard() noexcept
+{
+    cleanup();
+}
+operator std::string()
+{
+    return m_path;
+}
+private:
+std::string m_path;
+void cleanup() const noexcept
+{
+    try {
+        File::try_remove(m_path);
+        do_clean_dir(m_path + ".management", ".management");
+        if (File::is_dir(m_path + ".management"))
+            remove_dir(m_path + ".management");
+        File::try_remove(m_path + ".lock");
+    }
```

Comment:
I am not a fan of this information about locks, .management folders etc. being in multiple places. If the `SharedGroupTestPathGuard` can be reused somehow, it would be better.